### PR TITLE
Enhancement/Rewrite main-page using mui

### PR DESCRIPTION
**Pull request by InfectusProg**

### CODE REVIEW

File: src/components/pages/main-page/MainPage.styles.ts
Comment (Markusplay) [inline src/components/pages/main-page/MainPage.styles.ts:1]: remove comments

File: src/components/pages/main-page/MainPage.styles.ts [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+export const resources: SxProps<Theme> = {
+  display: 'flex',
+  alignItems: 'center',
+  justifyContent: 'center',
+  flexDirection: 'column',
+  margin: {
+    mobileSemiMedium: '97px 0 40px',
+    mobileMedium: '97px 0 40px',
+    tablet: '97px 0 127px',
+    desktopLarge: '97px 0 127px',
```
Comment (Markusplay) [inline src/components/pages/main-page/MainPage.styles.ts:226]: styles duplicates

File: src/components/pages/main-page/MainPage.styles.ts [hunk 2]
```diff
@@ -0,0 +1,258 @@
...
+};
+
+export const resources: SxProps<Theme> = {
+  display: 'flex',
+  alignItems: 'center',
+  justifyContent: 'center',
+  flexDirection: 'column',
+  margin: {
+    mobileSemiMedium: '97px 0 40px',
+    mobileMedium: '97px 0 40px',
```
Comment (Markusplay) [inline src/components/pages/main-page/MainPage.styles.ts:224]: styles duplicates

File: src/components/pages/main-page/MainPage.styles.ts [hunk 3]
```diff
@@ -0,0 +1,258 @@
...
+    tablet: '97px 0 127px',
+    desktopLarge: '97px 0 127px',
+  },
+  mobileMedium: {
+    marginBottom: '40px',
+  },
+};
+
+export const resourcesH3: SxProps<Theme> = {
+  fontWeight: '700',
```
Comment (Markusplay) [inline src/components/pages/main-page/MainPage.styles.ts:234]: you are using typography so don`t use fontWeight

File: src/components/pages/main-page/MainPage.tsx [hunk 4]
```diff
@@ -38,114 +39,115 @@ const MainPage: FC<MainPageProps> = ({ data }) => {
...
-          </div>
-        </div>
-        <div className={styles['resources']}>
-          <h3>Студентські ресурси</h3>
-          <div className={styles['resource-card-container']}>
-            <div className={styles['resources-cards']}>
+          </Box>
+        </Box>
+        <Box sx={stylesMUI.resources}>
+          <Typography variant={'h3Bold'} sx={stylesMUI.resourcesH3}>
```
Comment (Markusplay) [inline src/components/pages/main-page/MainPage.tsx:139]: if your font is changing by device with you don't need to set `variant={'h3Bold'}` as you have sx styles for it

File: src/components/pages/main-page/MainPage.tsx [hunk 5]
```diff
@@ -38,114 +39,115 @@ const MainPage: FC<MainPageProps> = ({ data }) => {
...
+            Студентські ресурси
+          </Typography>
+          <Box>
```
Comment (Markusplay) [inline src/components/pages/main-page/MainPage.tsx:142]: empty wrapper

File: src/components/pages/main-page/MainPage.tsx [hunk 6]
```diff
@@ -38,114 +39,115 @@ const MainPage: FC<MainPageProps> = ({ data }) => {
...
+                    text={'Обрати пріоритет'}
+                    disabled={false}
+                    variant={ButtonVariant.OUTLINE}
+                    size={ButtonSize.LARGE}
+                  />
+                </Link>
+              </Box>
+              <Box sx={stylesMUI.buttonTab}>
+                {!isLoggedIn && (
+                  <>
```
Comment (Markusplay) [inline src/components/pages/main-page/MainPage.tsx:84]: you don't need to wrap `Link` in `Fragment`

File: src/components/pages/main-page/MainPage.tsx [hunk 7]
```diff
@@ -38,114 +39,115 @@ const MainPage: FC<MainPageProps> = ({ data }) => {
...
+                    text={'Обрати пріоритет'}
+                    disabled={false}
+                    variant={ButtonVariant.OUTLINE}
+                    size={ButtonSize.MEDIUM}
+                  />
+                </Link>
+              </Box>
+              <Box sx={stylesMUI.buttonMob}>
+                {!isLoggedIn && (
+                  <>
```
Comment (Markusplay) [inline src/components/pages/main-page/MainPage.tsx:109]: you don't need to wrap `Link` in `Fragment`

File: src/components/pages/main-page/MainPage.tsx [hunk 8]
```diff
@@ -38,114 +39,115 @@ const MainPage: FC<MainPageProps> = ({ data }) => {
...
+                        size={ButtonSize.SMALL}
+                      />
+                    </Link>
+                  </>
+                )}
+                <Link href={'/priority'}>
+                  <Button
+                    sx={stylesMUI.buttons}
+                    text={'Обрати пріоритет'}
+                    disabled={false}
```
Comment (Markusplay) [inline src/components/pages/main-page/MainPage.tsx:126]: I guess disabled is false by default

File: src/components/pages/main-page/MainPage.tsx
Comment (Markusplay) [inline src/components/pages/main-page/MainPage.tsx:1]: I guess in `Button` component `disabled` is false by default

File: src/components/pages/main-page/MainPage.tsx [hunk 9]
```diff
@@ -38,114 +39,115 @@ const MainPage: FC<MainPageProps> = ({ data }) => {
...
     >
-      <div className={styles['main-page-content']}>
+      <Box sx={stylesMUI.mainPageContent}>
         {token && isReady && <TokenPopup token={token} />}
-        <div className={styles['header']}>
-          <div className={styles['header-info']}>
-            <div className={styles['header-info-content']}>
-              <h1 className={styles['title']}>Твій студентський портал</h1>
-              <p>
+        <Box sx={stylesMUI.header}>
```
Comment (Markusplay) [inline src/components/pages/main-page/MainPage.tsx:44]: rename styles cause it is not actually `header` for the page

File: src/components/pages/main-page/MainPage.tsx [hunk 10]
```diff
@@ -38,114 +39,115 @@ const MainPage: FC<MainPageProps> = ({ data }) => {
...
+          <Box sx={stylesMUI.headerInfo}>
```
Comment (Markusplay) [inline src/components/pages/main-page/MainPage.tsx:45]: same

File: src/components/pages/main-page/MainPage.tsx [hunk 11]
```diff
@@ -38,114 +39,115 @@ const MainPage: FC<MainPageProps> = ({ data }) => {
...
-          </div>
-          <div
-            className={cn(styles['build-image'], {
-              [styles['animate']]: !isIOS && !isSafari,
-            })}
-          >
+              </Typography>
+              <Box sx={stylesMUI.buttonDesk}>
+                {!isLoggedIn && (
+                  <>
```
Comment (Markusplay) [inline src/components/pages/main-page/MainPage.tsx:58]: you don't need to wrap Link in Fragment

File: src/components/pages/main-page/MainPage.tsx
Comment (Markusplay) [inline src/components/pages/main-page/MainPage.tsx:1]: You have duplicating in page - you have same buttons but with different containers for mobile, tablet and desktop.
Rows:
57-82
83-106
107-131
You need to create one container for them and use `useMediaQuery` for example to change size of buttons in page. Also change style values of their container for different sizes.

File: src/components/pages/main-page/MainPage.tsx [hunk 12]
```diff
@@ -38,114 +39,115 @@ const MainPage: FC<MainPageProps> = ({ data }) => {
...
         {token && isReady && <TokenPopup token={token} />}
-        <div className={styles['header']}>
-          <div className={styles['header-info']}>
-            <div className={styles['header-info-content']}>
-              <h1 className={styles['title']}>Твій студентський портал</h1>
-              <p>
+        <Box sx={stylesMUI.header}>
+          <Box sx={stylesMUI.headerInfo}>
+            <Box>
+              <Typography variant={'h2Bold'} sx={stylesMUI.title}>
```
Comment (Markusplay) [inline src/components/pages/main-page/MainPage.tsx:47]: you have typography in sx, so set typography there

File: src/components/pages/main-page/MainPage.tsx [hunk 13]
```diff
@@ -38,114 +39,115 @@ const MainPage: FC<MainPageProps> = ({ data }) => {
...
+                Твій студентський портал
+              </Typography>
+              <Typography variant={'body2'} paragraph sx={stylesMUI.titlePar}>
```
Comment (Markusplay) [inline src/components/pages/main-page/MainPage.tsx:50]: you have typography in sx, so set typography there

File: src/components/pages/main-page/MainPage.styles.ts [hunk 14]
```diff
@@ -0,0 +1,258 @@
...
+  height: {
+    tablet: 'fit-content',
+    desktop: 'fit-content',
+  },
+  position: 'relative',
+  zIndex: 1,
+  padding: {
+    mobile: '0 10px 0 17px',
+    mobileMedium: '0 40px 0 50px',
+    tablet: '0 40px 0 50px',
```
Comment (Markusplay) [inline src/components/pages/main-page/MainPage.styles.ts:46]: duplicates

File: src/components/pages/main-page/MainPage.styles.ts [hunk 15]
```diff
@@ -0,0 +1,258 @@
...
+  position: 'relative',
+  overflow: 'hidden',
+  flexDirection: {
+    mobile: 'column-reverse',
+    tablet: 'column-reverse',
+    desktop: 'row',
+  },
+  alignItems: {
+    mobile: 'flex-end',
+    tablet: 'flex-end',
```
Comment (Markusplay) [inline src/components/pages/main-page/MainPage.styles.ts:24]: duplicates

File: src/components/pages/main-page/MainPage.styles.ts [hunk 16]
```diff
@@ -0,0 +1,258 @@
...
+};
+
+export const header: SxProps<Theme> = {
+  width: '100%',
+  display: 'flex',
+  position: 'relative',
+  overflow: 'hidden',
+  flexDirection: {
+    mobile: 'column-reverse',
+    tablet: 'column-reverse',
```
Comment (Markusplay) [inline src/components/pages/main-page/MainPage.styles.ts:19]: duplicates

File: src/components/pages/main-page/MainPage.styles.ts [hunk 17]
```diff
@@ -0,0 +1,258 @@
...
+  display: 'flex',
+  flexDirection: 'column',
+  justifyContent: 'center',
+  width: {
+    tablet: '100%',
+    desktop: '30%',
+  },
+  height: {
+    tablet: 'fit-content',
+    desktop: 'fit-content',
```
Comment (Markusplay) [inline src/components/pages/main-page/MainPage.styles.ts:39]: duplicates

File: src/components/pages/main-page/MainPage.styles.ts [hunk 18]
```diff
@@ -0,0 +1,258 @@
...
+};
+
+export const title: SxProps<Theme> = {
+  typography: {
+    mobile: theme.typography.h4Bold,
+    tablet: theme.typography.h2Bold,
+  },
+  display: 'block',
+  position: 'relative',
+  zIndex: 100,
```
Comment (Markusplay) [inline src/components/pages/main-page/MainPage.styles.ts:58]: u sure that title needs so high zIndex ? :) Why not zIndex 2 e.g ?

File: src/components/pages/main-page/MainPage.styles.ts [hunk 19]
```diff
@@ -0,0 +1,258 @@
...
+  width: {
+    mobileMedium: '80%',
+    tablet: '100%',
+    desktop: '100%',
```
Comment (Markusplay) [inline src/components/pages/main-page/MainPage.styles.ts:62]: duplicates

File: src/components/pages/main-page/MainPage.styles.ts [hunk 20]
```diff
@@ -0,0 +1,258 @@
...
+    desktopSemiMedium: 'max-content',
+    desktopLarge: 'max-content',
```
Comment (Markusplay) [inline src/components/pages/main-page/MainPage.styles.ts:64]: duplicates

File: src/components/pages/main-page/MainPage.styles.ts [hunk 21]
```diff
@@ -0,0 +1,258 @@
...
+    mobileMedium: '-50px',
+    tablet: '-50px',
+    desktop: 0,
+  },
+};
+
+export const titlePar: SxProps<Theme> = {
+  padding: {
+    mobileSemiMedium: '0 17px 0 0',
+    mobileMedium: '0 17px 0 0',
```
Comment (Markusplay) [inline src/components/pages/main-page/MainPage.styles.ts:77]: duplicates

File: src/components/pages/main-page/MainPage.styles.ts [hunk 22]
```diff
@@ -0,0 +1,258 @@
...
+    tablet: 0,
+  },
+  margin: {
+    mobileSemiMedium: '0 0 36px',
+    mobileMedium: '0 0 44px',
+    tablet: '0 0 44px',
```
Comment (Markusplay) [inline src/components/pages/main-page/MainPage.styles.ts:83]: duplicates

File: src/components/pages/main-page/MainPage.styles.ts [hunk 23]
```diff
@@ -0,0 +1,258 @@
...
+    desktop: '-15px',
+    desktopSemiMedium: '-50px',
+    desktopMedium: '-15px',
+    desktopLarge: '13px',
+  },
+  svg: {
+    width: {
+      mobileSemiMedium: '100%',
+      mobileMedium: '80%',
+      tablet: '80%',
```
Comment (Markusplay) [inline src/components/pages/main-page/MainPage.styles.ts:123]: duplicates

File: src/components/pages/main-page/MainPage.styles.ts [hunk 24]
```diff
@@ -0,0 +1,258 @@
...
+      desktop: 'auto',
+    },
+    height: {
+      mobileSemiMedium: 'auto',
+      mobileMedium: 'auto',
```
Comment (Markusplay) [inline src/components/pages/main-page/MainPage.styles.ts:128]: duplicates

File: src/components/pages/main-page/MainPage.styles.ts [hunk 25]
```diff
@@ -0,0 +1,258 @@
...
+      tablet: '100%',
+      desktop: 'auto',
+      desktopSemiMedium: 'auto',
```
Comment (Markusplay) [inline src/components/pages/main-page/MainPage.styles.ts:131]: duplicates

File: src/components/pages/main-page/MainPage.styles.ts [hunk 26]
```diff
@@ -0,0 +1,258 @@
...
+    mobileSemiMedium: 'auto',
+  },
+  typography: {
+    mobile: theme.typography.body1Bold,
+    tablet: theme.typography.buttonBold,
+  },
+  borderRadius: {
+    mobile: '6px',
+    tablet: '8px',
+    desktopLarge: '8px',
```
Comment (Markusplay) [inline src/components/pages/main-page/MainPage.styles.ts:162]: duplicates

File: src/components/pages/main-page/MainPage.styles.ts [hunk 27]
```diff
@@ -0,0 +1,258 @@
...
+  },
+  padding: {
+    mobileSemiMedium: '8px 16px',
+    mobileMedium: '8px 16px',
+    tablet: '12px 24px',
+    desktopSemiMedium: '16px 32px',
+    desktop: '12px 24px',
```
Comment (Markusplay) [inline src/components/pages/main-page/MainPage.styles.ts:169]: duplicates

File: src/components/pages/main-page/MainPage.styles.ts [hunk 28]
```diff
@@ -0,0 +1,258 @@
...
+  },
+};
+export const buttonDesk: SxProps<Theme> = {
+  width: 'max-content',
+  display: {
+    mobile: 'none',
+    mobileMedium: 'none',
+    tablet: 'none',
```
Comment (Markusplay) [inline src/components/pages/main-page/MainPage.styles.ts:177]: duplicates

File: src/components/pages/main-page/MainPage.styles.ts [hunk 29]
```diff
@@ -0,0 +1,258 @@
...
+    desktop: 'flex',
+    desktopLarge: 'flex',
+  },
+  gap: '20px',
+};
+
+export const buttonTab: SxProps<Theme> = {
+  display: {
+    mobile: 'none',
+    mobileMedium: 'none',
```
Comment (Markusplay) [inline src/components/pages/main-page/MainPage.styles.ts:187]: duplicates

File: src/components/pages/main-page/MainPage.styles.ts [hunk 30]
```diff
@@ -0,0 +1,258 @@
...
+};
+
+export const buttonMob: SxProps<Theme> = {
+  typography: theme.typography.body1Bold,
+  // flexWrap: 'wrap',
+  gap: '14px',
+  width: '100%',
+  display: {
+    mobileSemiMedium: 'flex',
+    mobileMedium: 'flex',
```
Comment (Markusplay) [inline src/components/pages/main-page/MainPage.styles.ts:203]: duplicates

File: src/components/pages/main-page/MainPage.styles.ts [hunk 31]
```diff
@@ -0,0 +1,258 @@
...
+  mobileMedium: {
+    marginBottom: '40px',
+  },
+};
+
+export const resourcesH3: SxProps<Theme> = {
+  fontWeight: '700',
+  typography: {
+    mobileSemiMedium: theme.typography.h6Medium,
+    mobileMedium: theme.typography.h6Medium,
```
Comment (Markusplay) [inline src/components/pages/main-page/MainPage.styles.ts:237]: duplicates

File: src/components/pages/main-page/MainPage.styles.ts [hunk 32]
```diff
@@ -0,0 +1,258 @@
...
+  padding: {
+    mobile: '32px 17px',
+    tablet: '64px 0 0 0',
+  },
+  display: 'flex',
+  flexWrap: 'wrap',
+  justifyContent: 'center',
+  gap: {
+    mobileSemiMedium: '0.5rem',
+    mobileMedium: '0.5rem',
```
Comment (Markusplay) [inline src/components/pages/main-page/MainPage.styles.ts:252]: duplicates

**Review by Markusplay — CHANGES_REQUESTED** [review]

File: src/components/pages/main-page/MainPage.tsx (hunk 11 above)
Comment (InfectusProg) [reply src/components/pages/main-page/MainPage.tsx:58]: TS2657: JSX expressions must have one parent element. That`s why Fragment should stay for buttonDesk version (it has 2 children in the login-check)

Comment (InfectusProg) [reply src/components/pages/main-page/MainPage.tsx:58]: I left it dynamic version of buttons for the same reason

**Review by Markusplay — APPROVED** [review]

**Review by Yevheniy2020 — APPROVED** [review]

**Pull request merged**